### ISSUE TITLE
Outfits in cargo appear as "in stock" when not showing outfits in cargo

### ISSUE THREAD

**Issue by MessyMix** [endless-sky/endless-sky#2787]
I have 6 torpedos in my cargo hold.

However, I haven't ticked the box that says "show outfits in cargo". So it should only show the outfits that are in stock, and not the ones that are in my cargo hold. Therefore my 6 torpedos should be absent. However, they show up as being "in stock" instead! This is very misleading.

![image](https://user-images.githubusercontent.com/20450076/28457919-7166ee66-6e3a-11e7-83f2-639c2a394a38.png)

As you can see, it says that 6 torpedos are in stock, but they are actually in my cargo hold.

![image](https://user-images.githubusercontent.com/20450076/28458118-21a6eb28-6e3b-11e7-85ff-9e3c5a7dc253.png)

When I tick the box that shows only cargo, then it displays properly.

And before anyone asks, no, there aren't 6 in stock and 6 in cargo. there are only 6 in my cargo hold, but it's showing up as "in stock" when I haven't ticked the cargo box.

**Comment by Hadron1776** [comment]
May have something to do with this block of code:
```
	else if(!outfitter.Has(outfit))
	{
		// If not showing cargo, outfits in cargo count as "in stock."
		int count = player.Cargo().Get(outfit) + max(0, player.Stock(outfit));
		string message = (count > 0 ?
			"in stock: " + to_string(count) : "(not sold here)");
		Point pos = point + Point(
			OUTFIT_SIZE / 2 - 20 - font.Width(message),
			OUTFIT_SIZE / 2 - 24);
		font.Draw(message, pos, bright);
	}
```

**Comment by jafdy** [comment]
http://steamcommunity.com/app/404410/discussions/0/1458455461482174100/
People are getting confused by this behavior. It should probably be changed.

**Comment by VonZorch** [comment]
I've noticed it too and it is annoying but one can work around it.
A linked problem is when you have an item already in your hold and want to buy more it only shows the amount you have in cargo.
The simplest fix would probably be to show items in stock in one place and in cargo another, such as top and bottom of the items image.

**Comment by endless-sky** [comment]
Fixed in efbb384:

![cargo](https://user-images.githubusercontent.com/11450262/28482121-03964ef4-6e35-11e7-9062-cfdb5e9d9748.png)

Also made it so you can sell cargo even when not viewing cargo only, since that seems to be confusing people too.

**Comment by MessyMix** [comment]
Lol we pretty much went back to showing both cargo and in stock at the same time. Well, glad it's solved. :)
